src/server/ua_subscription_events.c

```diff
@@ -678,4 +678,216 @@ UA_Server_triggerEvent(UA_Server *server, const UA_NodeId eventNodeId,
+    if(eventFilter->selectClausesSize != 0) {
+        for(size_t i = 0; i < eventFilter->selectClausesSize; ++i) {
+            result[i] = UA_STATUSCODE_GOOD;
+            /* Checks if typedefenitionid or browsepath of any clause is NULL */
+            if(UA_NodeId_isNull(&eventFilter->selectClauses[i].typeDefinitionId)) {
+                result[i] = UA_STATUSCODE_BADTYPEDEFINITIONINVALID;
+                continue;
+            }
+            if(&eventFilter->selectClauses[i].browsePath[0] == NULL) {
+                result[i] = UA_STATUSCODE_BADBROWSENAMEINVALID;
+                continue;
+            }
+            /* Check if the eventType is a subtype of BaseEventType */
+            UA_NodeId baseEventTypeId = UA_NODEID_NUMERIC(0, UA_NS0ID_BASEEVENTTYPE);
+            if(!isNodeInTree_singleRef(
+                server, &eventFilter->selectClauses[i].typeDefinitionId,
+                &baseEventTypeId, UA_REFERENCETYPEINDEX_HASSUBTYPE)) {
+                result[i] = UA_STATUSCODE_BADTYPEDEFINITIONINVALID;
+                continue;
+            }
+            /* Check if attributeId is valid */
+            if(!((0 < eventFilter->selectClauses[i].attributeId) &&
+                 (eventFilter->selectClauses[i].attributeId < 28))) {
+                result[i] = UA_STATUSCODE_BADATTRIBUTEIDINVALID;
+                continue;
+            }
+            /* Check if browsePath contains null */
+            for(size_t j = 0; j < eventFilter->selectClauses[i].browsePathSize; ++j) {
+                if(UA_QualifiedName_isNull(
+                    &eventFilter->selectClauses[i].browsePath[j])) {
+                    result[i] = UA_STATUSCODE_BADBROWSENAMEINVALID;
+                    break;
+                }
+            }
+            if(result[i] != UA_STATUSCODE_GOOD)
+                continue;
+            /* Check if indexRange is defined */
+            if(!UA_String_equal(&eventFilter->selectClauses[i].indexRange,
+                                &UA_STRING_NULL)) {
+                /* Check if indexRange is parsable */
+                UA_NumericRange numericRange = UA_NUMERICRANGE("");
+                if(UA_NumericRange_parse(&numericRange,
+                                         eventFilter->selectClauses[i].indexRange) !=
+                   UA_STATUSCODE_GOOD) {
+                    result[i] = UA_STATUSCODE_BADINDEXRANGEINVALID;
+                    continue;
+                }
+                UA_free(numericRange.dimensions);
+                /* Check if attributeId is value */
+                if(eventFilter->selectClauses[i].attributeId != UA_ATTRIBUTEID_VALUE) {
+                    result[i] = UA_STATUSCODE_BADTYPEMISMATCH;
+                    continue;
+                }
+            }
+        }
+    }
+}
+
+UA_StatusCode
+UA_Event_staticWhereClauseValidation(UA_Server *server,
+                                     const UA_ContentFilter *contentFilter,
```

Comment:
Rename the arguments simply “filter” and “result”.
That makes all lines shorter and nicer to read.